src/shims/tls.rs

```diff
@@ -38,6 +38,9 @@ pub struct TlsData<'tcx> {
 
     /// Whether we are in the "destruct" phase, during which some operations are UB.
     dtors_running: HashSet<ThreadId>,
```

Comment:
Is this still needed? Seems like we could just check if `ThreadId` exists in `last_dtor_key`.

Reply:
Hm, okay maybe it is not entirely redundant, but I feel the two should be merged. Maybe into something like
```
dtors_running: FxHashMap<ThreadId, RunningDtorsState>
```
and then `RunningDtorsState` has all the state that we need to keep between subsequent `schedule_tls_dtors` calls.

Reply:
Oh also, could you have a test where a thread... let's say, spawns another thread and then joins on it, in a TLS dtor? Just to see this in action. :)